modules/table/src/main/java/org/apache/ignite/internal/table/distributed/replicator/PartitionReplicaListener.java

```diff
@@ -2250,9 +2256,12 @@ private CompletableFuture<ReplicaResult> processMultiEntryAction(ReadWriteMultiR
 
                         boolean insert = rowId == null;
 
-                        RowId rowId0 = insert ? new RowId(partId(), UUID.randomUUID()) : rowId;
+                        RowId rowId0 = insert
```

Comment:
Can we avoid taking locks for duplicate rows (in line 2275) ?

Reply:
Good catch. Fixed to reuse the lock and skip previous updates for the same key.